src/parser/helper/TreeParser.ts

```diff
@@ -22,4 +22,21 @@ export class TreeParser {
 
         return tree;
     }
+
+    static async getParseTreeAsync(parseFile: ParseFile): Promise<Tree> {
+        const cachedItem = TreeParser.cache.get(parseFile.filePath);
+        if (cachedItem !== undefined) {
+            return cachedItem;
+        }
+
+        const parser = new Parser();
+        parser.setLanguage(grammars.get(parseFile.language));
+
+        const sourceCode = (await fs.promises.readFile(parseFile.filePath)).toString();
```

Comment:
Instead of using `toString()` you can use the encoding option to directly read into a file. Otherwise we first read everything into a big buffer and then create a big string and Node.js has to copy everything.

Reply:
I have changed that to use the encoding option UTF-8, as according to the documentation, toString() without parameter uses UTF-8 anyway.